client/src/pages/queries/components/rename-query-modal.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect } from 'react'
+import { Form, Input, message, Modal } from 'antd'
+import { queryCache, useMutation } from 'react-query'
+import { SavedQuery, SavedQueryPage } from 'types/query'
+import { updateQuery } from '../queries-and-mutations'
+
+type RenameQueryModalProps = {
+  visible: boolean
+  onRequestClose: () => void
+  queryId: number
+  queryName: string
+}
+
+export default function RenameQueryModal({
+  visible,
+  onRequestClose,
+  queryId,
+  queryName,
+}: RenameQueryModalProps) {
+  const [form] = Form.useForm()
+
+  useEffect(
+    function setQueryName() {
+      form.setFieldsValue({ queryName })
+    },
+    [form, queryName],
+  )
+
+  const savedQueries: SavedQueryPage[] | undefined = queryCache.getQueryData([
+    'saved-queries',
+  ])
+
+  const savedQuery: SavedQuery | undefined = queryCache.getQueryData([
+    'saved-query',
+    queryId,
+  ])
+
+  const [updateQueryMutation] = useMutation(updateQuery, {
+    onMutate: (updatedQuery) => {
+      onRequestClose()
+      queryCache.setQueryData(
+        ['saved-queries'],
+        savedQueries?.map((page) => ({
```

Comment:
`setQueryData` also takes an `updater` function which provides old data. So, we can remove getting the `savedQueries` field from `queryCache`.